operator/internal/controller/auto/update_controller.go

```diff
@@ -621,6 +641,7 @@ type getResulter[T stream] struct {
 type result interface {
 	GetSummary() *agentpb.UpdateSummary
 	GetPermalink() string
+	GetIsStackLocked() bool
```

Comment:
Would prefer not to see `GetIsStackLocked` here.

The result interface is, I believe, intended to resemble the auto api.  I imagine that if the auto api had support for the "stack is locked" issue, it would be presented as an `err` with a status code or similar.  Can we make the GRPC interface be similar to that?